File: om/src/main/webapp/js/orgs/jobassignmenteditlist.js
/**
 * 
 */
function JobAssignmentEditList() {
	// newly added staff ID array on the position or the project role
	this.addedStaffIds = [];
	// removed staff ID array on the position or the project role
	this.removedStaffIds = [];

	// the position ID or project role ID
	this.positionId = null;
	// 0: position; 1: project role
	this.source = 0;
	// organization ID
	this.ownerId = null;

	// current staff array list on the position and the project role
	this.leftList = [];
	// current staff array list NOT on the position and the project role
	this.rightList = [];

	// the description for changed assignments
	this.description = null;

};

JobAssignmentEditList.prototype.parseFromJSON = function(json, ownerid) {
	this.positionId = json.positionId;
	if (json.leftList.length > 0) {
		this.leftList = [];
		for (var i = 0; i < json.leftList.length; i++) {
			var member = new JobAssignment();
			member.parseFromJSON(json.leftList[i], ownerid);
			this.leftList.push(member);
		}
	}
	if (json.rightList.length > 0) {
		this.rightList = [];
		for (var i = 0; i < json.rightList.length; i++) {
			var member = new JobAssignment();
			member.parseFromJSON(json.rightList[i], ownerid);
			this.rightList.push(member);
		}
	}
	this.ownerId = ownerid;
};

JobAssignmentEditList.prototype.findMember = function(id) {
	var gm = null;
	for (var i = 0; i < this.leftList.length; i++) {
		if (this.leftList[i].staffId == id) {
			gm = this.leftList[i];
		}
	}
	if (gm == null) {
		for (var i = 0; i < this.rightList.length; i++) {
			if (this.rightList[i].staffId == id) {
				gm = this.rightList[i];
			}
		}
	}
	return gm;
};

JobAssignmentEditList.prototype.searchLeftList = function(condition) {
	var result = [];
	var list = this.leftList;
	for (var i = 0; i < list.length; i++) {
		if ((list[i].staffCode != null && list[i].staffCode.search(condition) >= 0)
				|| (list[i].staffFullName != null && list[i].staffFullName
						.search(condition) >= 0)) {
			result.push(list[i]);
		}
	}
	return result;
};

JobAssignmentEditList.prototype.searchRightList = function(condition) {
	var result = [];
	var list = this.rightList;
	for (var i = 0; i < list.length; i++) {
		if ((list[i].staffCode != null && list[i].staffCode.search(condition) >= 0)
				|| (list[i].staffFullName != null && list[i].staffFullName
						.search(condition) >= 0)) {
			result.push(list[i]);
		}
	}
	return result;
};

JobAssignmentEditList.prototype.clearChangedStaffIds = function() {
	var t1 = [];
	for (var i = 0; i < this.addedStaffIds.length; i++) {
		if (!this.existLeftStaffIds(this.addedStaffIds[i])
				&& !this.existRemovedStaffIds(this.addedStaffIds[i])) {
			t1.push(this.addedStaffIds[i]);
		}
	}
	// console.log(t1);
	this.addedStaffIds = t1;
	var t2 = [];
	for (var i = 0; i < this.removedStaffIds.length; i++) {
		if (this.existLeftStaffIds(this.removedStaffIds[i])) {
			t2.push(this.removedStaffIds[i]);
		}
	}
	// console.log(t2);
	this.removedStaffIds = t2;
};

JobAssignmentEditList.prototype.existLeftStaffIds = function(id) {
	for (var j = 0; j < this.leftList.length; j++) {
		if (this.leftList[j].staffId == id) {
			return true;
		}
	}
	return false;
};

JobAssignmentEditList.prototype.existRemovedStaffIds = function(id) {
	for (var j = 0; j < this.removedStaffIds.length; j++) {
		if (this.removedStaffIds[j] == id) {
			return true;
		}
	}
	return false;
};

File: core.organization/src/main/java/com/cloudibpm/core/job/JobAssignmentEditList.java
/**
 * 
 */
package com.cloudibpm.core.job;

import java.io.Serializable;

/**
 * @author dev
 *
 */
public class JobAssignmentEditList implements Serializable {

	/**
	 * serialVersionUID
	 */
	private static final long serialVersionUID = -4549090663315271876L;
	// newly added staff ID array on the position or the project role
	private String[] addedStaffIds = null;
	// removed staff ID array on the position or the project role
	private String[] removedStaffIds = null;

	// the position ID or project role ID
	private String positionId = null;
	private String staffId = null;
	// 0 means that the positionId is a position ID, 1 means that positionId is
	// a project role ID.
	private int source = 0;
	// organization ID
	private String ownerId = null;

	// current staff array list on the position and the project role
	private JobAssignment[] leftList = null;
	// current staff array list NOT on the position and the project role
	private JobAssignment[] rightList = null;

	// the description for changed assignments
	private String description = null;

	/**
	 * 
	 */
	public String getStaffId() {
		return staffId;
	}

	public void setStaffId(String staffId) {
		this.staffId = staffId;
	}
	
	public JobAssignmentEditList() {
	}

	public String[] getAddedStaffIds() {
		return addedStaffIds;
	}

	public void setAddedStaffIds(String[] addedStaffIds) {
		this.addedStaffIds = addedStaffIds;
	}

	public String[] getRemovedStaffIds() {
		return removedStaffIds;
	}

	public void setRemovedStaffIds(String[] removedStaffIds) {
		this.removedStaffIds = removedStaffIds;
	}

	public String getPositionId() {
		return positionId;
	}

	public void setPositionId(String positionId) {
		this.positionId = positionId;
	}

	public String getOwnerId() {
		return ownerId;
	}

	public void setOwnerId(String ownerId) {
		this.ownerId = ownerId;
	}

	public JobAssignment[] getLeftList() {
		return leftList;
	}

	public void setLeftList(JobAssignment[] leftList) {
		this.leftList = leftList;
	}

	public JobAssignment[] getRightList() {
		return rightList;
	}

	public void setRightList(JobAssignment[] rightList) {
		this.rightList = rightList;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public int getSource() {
		return source;
	}

	public void setSource(int source) {
		this.source = source;
	}

}
